apps/coscrad-frontend/src/components/resources/factories/table-view-categorizable-index-presenter-factory.tsx

```diff
@@ -0,0 +1,76 @@
+import {
+    CategorizableType,
+    IIndexQueryResult,
+    ISpatialFeatureViewModel,
+} from '@coscrad/api-interfaces';
+import { BibliographicReferenceIndexPresenter } from '../bibliographic-references/bibliographic-reference-index.presenter';
+import { BookIndexPresenter } from '../books/book-index.presenter';
+import { MediaItemIndexContainer } from '../media-items';
+import { PhotographIndexPresenter } from '../photographs/photograph-index.presenter';
+import { SongIndexPresenter } from '../songs/song-index.presenter';
+import { CoscradLeafletMap } from '../spatial-features/leaflet';
+import { SpatialFeatureIndexPresenter } from '../spatial-features/spatial-feature-index.presenter';
+import { TermIndexPresenter } from '../terms/term-index.presenter';
+import { TranscribedAudioIndexPresenter } from '../transcribed-audio/transcribed-audio-index.presenter';
+import { VocabularyListIndexPresenter } from '../vocabulary-lists/vocabulary-list-index.presenter';
+import {
+    CategorizableIndexPresenterFactory,
+    IndexPresenter,
+} from './categorizable-index-presneter-factory.interface';
+import { thumbnailCategorizableDetailPresenterFactory } from './thumbnail-categorizable-detail-presenter-factory';
+
+/**
+ * TODO Find a better way to inject the dependencies
+ *
+ * Note- this breaks! Fix me before merging in!
+ */
+const ConcreteSpatialFeaturePresenter = (
+    result: IIndexQueryResult<ISpatialFeatureViewModel>
+): JSX.Element => (
+    <SpatialFeatureIndexPresenter
+        MapComponent={CoscradLeafletMap}
+        DetailPresenter={thumbnailCategorizableDetailPresenterFactory(
+            CategorizableType.spatialFeature
+        )}
+        {...result}
+    />
+);
+
+export const tableViewCategorizableIndexPresenterFactory: CategorizableIndexPresenterFactory = <
+    T extends CategorizableType
+>(
+    categorizableType: T
+): IndexPresenter<T> => {
+    switch (categorizableType) {
```

Comment:
Keep considering how to reduce the number of such switches \ lookup tables on categorizable type.

Reply:
Good goal, hard to get there...

Reply:
No- it's just run of the mill polymorphism. The reason it is awkward is that we are exclusively using functional programming techniques. That doesn't mean that we can't have a single switch, though. I see it as the key to being able to keep presentational logic orthogonal to state dependent logic.